### PR TITLE
Fix #683: Support static array in deserializer

### LINKED ISSUE #683: Deserializer does not work with fixed-length arrays
https://github.com/bpfkorea/agora/blob/d8154e24ed91844e7730ed646b5c3913cb5ec8fc/source/agora/common/Serializer.d#L548-L553

It's because of the `.array` call.

### PR DESCRIPTION
**Pull request by Geod24**
Also two minor commit I had laying around.

### CODE REVIEW

File: source/agora/common/Serializer.d [hunk 1]
```diff
@@ -523,6 +531,18 @@ public T deserializeFull (T) (scope DeserializeDg dg,
...
+    // Static array needs to be handled before arrays
+    // Note: This might be optimizable for small types (char, ubyte)
+    else static if (is(T : E[N], E, size_t N))
+    {
+        E deserializeEntry (size_t idx) ()
+        {
+            return deserializeFull!E(dg, opts);
+        }
+        // Note: This does not allocate because `staticMap` yields a tuple
+        return [ staticMap!(deserializeEntry, StaticIota!N) ];
```
Comment (Geod24) [inline source/agora/common/Serializer.d]: Note to self: D truly is marvelous

Comment (AndrejMitrovic) [reply source/agora/common/Serializer.d]:
```suggestion
        return [ Repeat!(N, deserializeEntry) ];
```

Sorry!

Comment (AndrejMitrovic) [reply source/agora/common/Serializer.d]: And change `deserializeEntry` to remove the template parameter.

Comment (Geod24) [reply source/agora/common/Serializer.d]: Oh super nice !

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**